### PR TITLE
[app] Use single, embedded DB for development

### PR DESCRIPTION
**Pull request by fkling**
**tl;dr:** Updating `PGHOST` et al. to point to the embedded DB didn't affect the relevant env vars for codintel and codesights because those are explicitly set by `sg` (see `sg.config.yaml`), causing the DB configuration to go out of sync. 

---

This was a weird/surprising issue. I wasn't able to run `sg start app` without having an external DB running. This was surprising especially since #48020 should have enabled that. 
Contrary to what I said in #50859, the embedded DB _is_ used, but only for the frontend. CodeIntel and CodeInsights require an external DB.
Why? Because by default `sg.config.yaml` sets the `CODEINTEL_PG*` and `CODEINSIGHTS_PG*` variables to the same values as `PG*`. Later the embedded DB logic updates the `PG*` variables but that doesn't affect the other ones because they have been set explicitly (`setDefaultEnv` only updates the env var if it's empty).

There are multiple ways to fix this:
- We could explicitly clear the env vars in sg.config.yaml, so that they are set when initializing the embedded DB.
- We can always overwrite the codeintel and codeinsights env vars.

I chose the latter approach here because that seems more in line with the purpose of `useSinglePostgreSQLDatabase`. At the moment it's not obvious that the different services might use different databases even after calling this functions.

Additional notes:

- This only affects `sg start` because it explicitly sets the codeintel and codeinsights env vars, causing `useSingle...` to "fail".
- It probably never worked on macOS either. Instead devs had probably a PostgreSQL DB running.
- With this change it's not possible anymore to point codinsights and codeintel to different DBs in dev mode, but I don't know it that's actually important.

## Test plan
`sg start app` runs successfully without having another prostgres service running.

### CODE REVIEW

**Comment by burmudar** [comment]
@fkling Nice! I have a plan to let app read it's environment variables from a app.env file, otherwise like you (and I) have experienced now different variables interfere and it is extremely difficult to track down

**Review by burmudar — APPROVED** [review]
Small test issue it seems - I'll take a look

**Comment by fkling** [comment]
Thank you @burmudar, I appreciate it!

**Comment by fkling** [comment]
Of course if this file is also used by non-app normal code then this wouldn't be the right approach I guess... but I assume it isn't?
